test/functional/test_framework/test_node.py

```diff
@@ -307,7 +307,7 @@ def wait_until_stopped(self, timeout=BITCOIND_PROC_WAIT_TIMEOUT):
         wait_until(self.is_node_stopped, timeout=timeout)
 
     @contextlib.contextmanager
-    def assert_debug_log(self, expected_msgs, timeout=2):
+    def assert_debug_log(self, expected_msgs, unexpected_msgs=[], timeout=2):
```

Comment:
Do you really want `unexpected_msgs` to preserve parameter state between calls? :-)

```
>>> def foobar(foo, bar=[]):
...     bar.append(foo)
...     print(bar)
...
>>> foobar("hello")
['hello']
>>> foobar("hello")
['hello', 'hello']
>>> foobar("hello")
['hello', 'hello', 'hello']
```

FWIW I think we should use a linter to guard against this Python gotcha :-)

Reply:
Great catch, I had no idea! I fixed this in #16115 (where it was introduced) and force-pushed rebased this PR onto that branch.

Reply:
@practicalswift, I don't think this was an actual problem. I added a debug print and `unexpected_msgs` didn't include previous values. I think it's not broken because this function, `assert_debug_log()`, doesn't modify the variable (it's not mutable). Here's an extremely simplified version of this function and some sample calls to it:
```
def assert_debug_log(unexpected_msgs=[]):
    print(unexpected_msgs)
    for unexpected_msg in unexpected_msgs:
        print('  ', unexpected_msg)

assert_debug_log()
assert_debug_log(["foo"])
assert_debug_log(unexpected_msgs=["foo"])
assert_debug_log("bar") # not actually called this way but just curious
assert_debug_log()
```
The output:
```
[]
['foo']
   foo
['foo']
   foo
bar
   b
   a
   r
[]
```
What prompted me to investigate this further is that I wondered if any of our existing python code has this problem. I found just one instance, which is in `wallet_importmulti.py`:
```
def test_importmulti(self, req, success, error_code=None, error_message=None, warnings=[]):
```
But adding a debug print to that function showed everything was okay. I believe that's because `warnings` isn't modified in that function.